splink/cluster_metrics.py

```diff
@@ -1,53 +1,123 @@
+from typing import Dict, List
+
 from splink.splink_dataframe import SplinkDataFrame
 
 
-def _size_density_sql(
+def _node_degree_sql(
     df_predict: SplinkDataFrame,
     df_clustered: SplinkDataFrame,
-    threshold_match_probability: float,
     composite_uid_edges_l: str,
+    composite_uid_edges_r: str,
     composite_uid_clusters: str,
-):
-    """Generates sql for computing cluster size and density at a given threshold.
+    threshold_match_probability: float,
+) -> List[Dict[str, str]]:
+    """
+    Generates sql for computing node degree per node, at a given edge threshold.
+
+    This is includes nodes with no edges, as identified via the clusters table.
 
     Args:
         df_predict (SplinkDataFrame): The results of `linker.predict()`.
         df_clustered (SplinkDataFrame): The outputs of
                 `linker.cluster_pairwise_predictions_at_threshold()`.
+        composite_uid_edges_l (str): unique id for left-hand edges.
+        composite_uid_edges_r (str): unique id for right-hand edges.
+        composite_uid_clusters (str): unique id for clusters.
         threshold_match_probability (float): Filter the pairwise match
             predictions to include only pairwise comparisons with a
             match_probability above this threshold.
-        composite_uid_edges_l (str): unique id for left-hand edges.
-        composite_uid_clusters (str): unique id for clusters.
-
 
     Returns:
-        sql string for computing cluster size and density
+        array of dicts, with sql string and output name
+        for computing cluster size and density
     """
-
     sqls = []
-    # Count edges per node at or above a given match probability
+    edges_tn = df_predict.physical_name
+    clusters_tn = df_clustered.physical_name
```

Comment:
This might come down to person preference but I find it easier to follow what's going on in the sql if e.g. `df_predict.physical_name` is included explicitly rather than an alias e.g. `edges_tn`. It also wasn't immediately obvious to me what `tn` stood for (although this could be a convention I'm just not familiar with 🙃)

Reply:
That makes sense - think this was left from a version where the SQL was more complicated so these tables were getting used in several places.